src/main/java/com/ocelotslovebirds/birdhaus/blocks/BirdhouseBlockEntity.java

```diff
@@ -70,14 +84,41 @@ public void load(CompoundTag tag) {
      */
 
     public void tickServer() {
+        // Find all parrots in the vicinity
+        List<Parrot> parrots = this.level.getEntitiesOfClass(Parrot.class, this.birdHouseBB);
         if (counter > 0) {
             counter--;
             setChanged();
+
+            // Add the new goal to the parrots if they dont already have it
+            for (Parrot temp:parrots) {
+                // Check to see if they already have the goal applied. If not, apply it.
+                if (!temp.goalSelector.getAvailableGoals().stream()
+                    .anyMatch(wg -> wg.getGoal() instanceof HangAroundBirdhouseGoal)) {
+                    temp.goalSelector.addGoal(0, new HangAroundBirdhouseGoal(temp, 1.0, 60, false, this.getBlockPos()));
+                }
+            }
```

Comment:
Well hopefully `tickServer` is something that is called synchronously, i.e. one birdhouse at a time, otherwise this code block is not thread safe 😹 

I'm willing to pretend that it is safe for now, but I think we should include a comment about this in the code.